server/src/daemon/service.rs

```diff
@@ -0,0 +1,265 @@
+//  org.freedesktop.Secret.Service
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+    time::SystemTime,
+};
+
+use oo7::{
+    dbus::{
+        api::{Properties, SecretInner},
+        Algorithm,
+    },
+    portal::{Item, Keyring},
+    Key,
+};
+use zbus::{
+    proxy::ProxyDefault,
+    zvariant::{ObjectPath, OwnedObjectPath, OwnedValue, Value},
+    ObjectServer, SignalContext,
+};
+
+use super::{
+    collection::Collection, error::ServiceError, prompt::Prompt, session::Session, Result,
+};
+
+#[derive(Debug)]
+pub struct Service {
+    collections: Vec<Collection>,
+    keyring: Arc<Keyring>,
+    cnx: Mutex<Option<zbus::Connection>>,
```

Comment:
I was thinking,

```
diff --git a/server/src/daemon/service.rs b/server/src/daemon/service.rs
index d757670..e221b42 100644
--- a/server/src/daemon/service.rs
+++ b/server/src/daemon/service.rs
@@ -21,14 +21,22 @@ use zbus::{
 };
 
 use super::{
-    collection::Collection, error::ServiceError, prompt::Prompt, session::Session, Result,
+    collection::Collection, error::ServiceError, item, prompt::Prompt, session::Session, Result,
 };
 
+#[derive(Debug)]
+pub struct ServiceManager {
+    collections: HashMap<OwnedObjectPath, Collection>,
+    items: HashMap<OwnedObjectPath, item::Item>,
+    sessions: HashMap<OwnedObjectPath, Session>,
+}
+
 #[derive(Debug)]
 pub struct Service {
     collections: Vec<Collection>,
     keyring: Arc<Keyring>,
     cnx: Mutex<Option<zbus::Connection>>,
+    manager: Arc<ServiceManager>,
 }
 
 #[zbus::interface(name = "org.freedesktop.Secret.Service")]
```

Reply:
This might make sense yes, although, I guess the collections are not supposed to be stored there or the items. As you don't want to mix items from different collections. 

So maybe just store the sessions there for now and see what requirements you end up needing with time.

Reply:
got it, thanks!